pkg/apis/numaplane/v1alpha1/progressive_status_types.go

```diff
@@ -24,6 +24,7 @@ import (
 )
 
 type AssessmentResult string
+type AssessmentFailureReason string
```

Comment:
is there any reason to have this as its own type? Were you just imitating the fact that `AssessmentResult` is its own type? The reason we have that as its own type is because it has a set of very specific values, but I don't think this does, right?

Reply:
For keeping it consistent with `AssessmentResult`, I have added this type. I don't have any solid reason, I'll remove it for now.